amethyst_network/src/network_socket.rs

```diff
@@ -162,50 +137,60 @@ where
 {
     type SystemData = (WriteStorage<'a, NetConnection<E>>);
 
-    fn setup(&mut self, res: &mut Resources) {
-        Self::SystemData::setup(res);
-    }
-
     fn run(&mut self, mut net_connections: Self::SystemData) {
         for net_connection in (&mut net_connections).join() {
-            let target = net_connection.target;
+            let target = net_connection.target_receiver;
 
             if net_connection.state == ConnectionState::Connected
                 || net_connection.state == ConnectionState::Connecting
             {
-                self.tx
+                self.transport_sender
                     .send(InternalSocketEvent::SendEvents {
                         target,
                         events: net_connection.send_buffer_early_read().cloned().collect(),
                     })
                     .expect("Unreachable: Channel will be alive until a stop event is sent");
             } else if net_connection.state == ConnectionState::Disconnected {
-                self.tx
+                self.transport_sender
                     .send(InternalSocketEvent::Stop)
                     .expect("Already sent a stop event to the channel");
             }
         }
 
-        for raw_event in self.rx.try_iter() {
+        let mut counter = 0;
+        for raw_event in self.transport_receiver.try_iter() {
             // Get the NetConnection from the source
             for net_connection in (&mut net_connections).join() {
-                // We found the origin
-                if net_connection.target == raw_event.source {
+                if net_connection.target_sender == raw_event.addr() {
                     // Get the event
-                    let net_event = deserialize_event::<E>(raw_event.data.as_slice());
+                    let net_event = deserialize_event::<E>(raw_event.payload());
                     match net_event {
                         Ok(ev) => {
                             net_connection.receive_buffer.single_write(ev);
                         }
                         Err(e) => error!(
                             "Failed to deserialize an incoming network event: {} From source: {:?}",
-                            e, raw_event.source
+                            e,
+                            raw_event.addr()
                         ),
                     }
                 } else {
                     warn!("Received packet from unknown source");
                 }
             }
+
+            // this will prevent our system to be stuck in the iterator.
+            // After 10000 packets we will continue and leaf the other packets for the next run.
+            // eventually some congestion prevention should be done.
+            if counter >= 10000 {
```

Comment:
Can we make this a constant somewhere? Or even better, something we can set via the config?